substrate/frame/nomination-pools/src/lib.rs

```diff
@@ -2265,7 +2244,8 @@ pub mod pallet {
 
 			// Before calculating the `balance_to_unbond`, we call withdraw unbonded to ensure the
 			// `transferrable_balance` is correct.
-			let stash_killed = bonded_pool.withdraw_from_staking(num_slashing_spans)?;
+			let stash_killed =
```

Comment:
Can we not update the TVL here, instead of needing the `on_withdraw`?

Reply:
That's what the `fn withdraw_from_staking` was doing. However that's insufficient because staking apply a withdraw unbonded without request from pools. If that happens, the TVL in the pool is not updated.

This is the culprit code on the staking side: https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/staking/src/pallet/mod.rs#L1029-L1039.